lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -412,6 +412,37 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Given string \p origName, generate a new name if it conflicts with any
+/// keyword or any other name in the set \p recordNames. Use the int \p
+/// nextGeneratedNameID as a counter for suffix. Update the \p recordNames with
+/// the generated name and return an iterator corresponding to the new entry.
+llvm::StringMapIterator<llvm::NoneType>
+resolveKeywordConflict(StringRef origName, llvm::StringSet<> &recordNames,
```

Comment:
I'd recommend changing this to return a `StringRef` directly.  One subtle but important thing about `StringSet` is that the entries in it have a stable pointer as soon as it is in the set.

Reply:
If we return the ```StringRef```, how to handle the following case, where the ```nameTable``` key is the ```StringMapItertor```
```
  auto iter = resolveKeywordConflict(name, usedNames, nextGeneratedNameID);
  if (valueOrOp)
    nameTable[valueOrOp] = &*iter;

```
We can either change the ```nameTable``` key type to ```StringRef```  or query the ```StringSet``` again to get the iterator.

Reply:
I'm not sure what you mean.  The key of nameTable seems fine, do you mean the value?  The value should just be a stringref

Reply:
Sorry, I meant the value, will change it to StringRef  ```llvm::DenseMap<ValueOrOp, llvm::StringMapEntry<llvm::NoneType> *> nameTable;
```

Reply:
Thanks for reviewing, made the following changes
1. ```resolveKeywordConflict``` now inserts into the ``StringSet``` and returns ```StringRef```
2.  Now only one lookup into ```recordNames```. 
3. ```nameTable; ``` was originally storing the iterator, but modified it to ```StringRef```